src/react-apps/applications/service-development/src/features/deploy/containers/deployToTestContainer.tsx

```diff
@@ -0,0 +1,260 @@
+  },
+  aboutServicePlaceholderStyle: {
+    [theme.breakpoints.up('md')]: {
+      borderLeft: '1px solid ' + theme.altinnPalette.primary.greyMedium,
+      paddingLeft: 12,
+    },
+    marginTop: 24,
+  },
+  mainGridStyle: {
+    maxWidth: 1340,
+  },
+});
+
+export enum inSyncStatus {
+  ahead = 'ahead',
+  behind = 'behind',
+  ready = 'ready',
+}
+
+export interface IDeployToTestContainerProps extends WithStyles<typeof styles> {
+  compileStatus: any;
+  compileStatusUniqueFilenames: [];
+  deploymentList: any;
+  deployStatus: any;
+  language: any;
+  masterRepoStatus: any;
+  repoStatus: any;
+}
+
+export class DeployToTestContainer extends
+  React.Component<IDeployToTestContainerProps> {
+
+  public interval: any = null;
+
+  constructor(_props: IDeployToTestContainerProps) {
+    super(_props);
+  }
+
+  public componentDidMount() {
+    const altinnWindow: any = window;
+    const { org, service } = altinnWindow;
+    DeployActionDispatcher.fetchDeployments(environment, org, service);
+    DeployActionDispatcher.fetchMasterRepoStatus(org, service);
+    DeployActionDispatcher.fetchCompileStatus(org, service);
+    window.postMessage(postMessages.forceRepoStatusCheck, window.location.href);
+
+    // If deployment has started but not finished, start the fetchDeploymentStatusInterval
+    if (this.props.deployStatus[environment].deployStartedSuccess === true &&
+      !this.props.deployStatus[environment].result.finishTime
+    ) {
+      this.fetchDeploymentStatusInterval(environment);
+
+      // Else if deployment is finished, then reset deploymentStatus
+    } else if (this.isDeployFinished(environment)) {
+      DeployActionDispatcher.resetDeploymentStatus(environment);
+    }
+
+  }
+
+  public componentWillUnmount() {
+    clearInterval(this.interval);
+  }
+
+  public isDeployFinished = (letEnv: string): boolean => {
+    if (this.props.deployStatus[letEnv].deployStartedSuccess === true &&
+      this.props.deployStatus[letEnv].result.finishTime) {
+      return true;
+    } else {
+      return false;
+    }
+  }
+
+  public returnInSyncStatus = (repoStatus: any): any => {
+    if (repoStatus.contentStatus) {
+      if (repoStatus.contentStatus.length > 0) {
+        return inSyncStatus.ahead;
+      } else if (repoStatus.behindBy > 0) {
+        return inSyncStatus.behind;
+      } else {
+        return inSyncStatus.ready;
+      }
+    } else {
+      return null;
+    }
+  }
+
+  public isMasterRepoAndDeployInSync = (letEnv: string, masterRepoStatus: any, deploymentList: any): any => {
+    const image = deploymentList[letEnv].items[0].spec.template.spec.containers[0].image;
+    const imageTag = image.split(':')[1];
+    if (masterRepoStatus !== null && masterRepoStatus.commit.id === imageTag) {
+      return true;
+    } else {
+      return false;
+    }
+  }
+
+  public startDeployment = (letEnv: string) => {
+    const altinnWindow: IAltinnWindow = window as IAltinnWindow;
+    const { org, service } = altinnWindow;
```

Comment:
Tidier with deconstruction on the same line.
const { org, service } = window as IAltinnWIndow;

Reply:
Fixed.